tools/flowc/backends/nim/runtime.nim

```diff
@@ -479,7 +475,12 @@ proc rt_to_string_quot*(f: Flow): RtString =
   of rtVoid:   return rt_to_string()
   of rtBool:   return rt_to_string(f.bool_v)
   of rtInt:    return rt_to_string(f.int_v)
-  of rtDouble: return rt_to_string(f.double_v)
+  of rtDouble:
+    # NOTE: toString(42.0) == "42.0" BUT cast(42.0: double -> string) == "42"
+    var x = formatFloat(f.double_v)
+    x.trimZeros()
+    if not x.contains('.'): x.add(".0")
```

Comment:
I think it should be if it does not contain "." AND not contain "e".

Reply:
It happes that to check just for '.' is sufficient. Citation: "The exponential ("E") format specifier converts a number to a string of the form "-d.ddd…E+ddd" or "-d.ddd…e+ddd", where each "d" indicates a digit (0-9). The string starts with a minus sign if the number is negative. Exactly one digit always precedes the decimal point." So, in case of 'e' sign the dot must be also in the string.